From 1f10fa2e06e46c18bba8370d9912555faaf0cd3f Mon Sep 17 00:00:00 2001
From: link <link@inventec.com>
Date: Wed, 2 Aug 2023 15:07:23 +0800
Subject: [PATCH] Bug 1028 - [SW][Transformers-nuv][kernel] - Add
 SAPPHIRERAPIDS for intel peci

Symptom/Reason:
    We sync the peci linux driver form Intel's public github.
    It didn't support sapphire rapids(Eagle stream).
    We need to patch the drivers.

Root Cause:
    N/A

Solution/Change:
    Add sapphirerapids support for peci.

Entry Test:

sysadmin@transformers-nuv:/sys/class/hwmon/hwmon7# ipmitool sdr list
DTS_CPU1         | 45 degrees C      | ok
Die_CPU1         | 45 degrees C      | ok
....
DIMM_A1_CPU1     | 30 degrees C      | ok
---
 drivers/hwmon/peci/peci-dimmtemp.c    | 6 ++++++
 include/linux/mfd/intel-peci-client.h | 6 ++++++
 2 files changed, 12 insertions(+)

diff --git a/drivers/hwmon/peci/peci-dimmtemp.c b/drivers/hwmon/peci/peci-dimmtemp.c
index 456991169716..eaa56e765aa4 100644
--- a/drivers/hwmon/peci/peci-dimmtemp.c
+++ b/drivers/hwmon/peci/peci-dimmtemp.c
@@ -43,6 +43,7 @@ static const u8 support_model[] = {
 	INTEL_FAM6_SKYLAKE_XD,
 	INTEL_FAM6_ICELAKE_X,
 	INTEL_FAM6_ICELAKE_XD,
+	INTEL_FAM6_SAPPHIRERAPIDS_X,
 };
 
 static inline int read_ddr_dimm_temp_config(struct peci_dimmtemp *priv,
@@ -267,6 +268,11 @@ static int get_dimm_temp(struct peci_dimmtemp *priv, int dimm_no)
 		priv->temp_max[dimm_no] = rp_msg.pci_config[1] * 1000;
 		priv->temp_crit[dimm_no] = rp_msg.pci_config[2] * 1000;
 		break;
+	case INTEL_FAM6_SAPPHIRERAPIDS_X:
+		/*<TBD> not sure how to get temp config hard code here as Icelake patch*/
+		priv->temp_max[dimm_no] = 85 * 1000; //85C
+		priv->temp_crit[dimm_no] = 90 * 1000; //90C
+		break;
 	default:
 		return -EOPNOTSUPP;
 	}
diff --git a/include/linux/mfd/intel-peci-client.h b/include/linux/mfd/intel-peci-client.h
index 0a069b87f733..723ebac687df 100644
--- a/include/linux/mfd/intel-peci-client.h
+++ b/include/linux/mfd/intel-peci-client.h
@@ -20,6 +20,7 @@
 #define INTEL_FAM6_SKYLAKE_XD		0x56
 #define INTEL_FAM6_ICELAKE_X		0x6A
 #define INTEL_FAM6_ICELAKE_XD		0x6C
+#define INTEL_FAM6_SAPPHIRERAPIDS_X	0x8F
 #endif
 
 #define INTEL_FAM6             6 /* P6 (Pentium Pro and later) */
@@ -48,6 +49,11 @@
 #define CHAN_RANK_MAX_ON_ICXD  4  /* Max number of channel ranks on Icelake D */
 #define DIMM_IDX_MAX_ON_ICXD   2  /* Max DIMM index per channel on Icelake D */
 
+/*SAPPHIRERAPIDS*/
+#define CORE_MASK_BITS_ON_SAPPHIRERAPIDS 50 /* Max number of cores */
+#define CHAN_RANK_MAX_ON_SAPPHIRERAPIDS  8  /* Max number of channel ranks*/
+#define DIMM_IDX_MAX_ON_SAPPHIRERAPIDS   2  /* Max DIMM index per channel*/
+
 #define CORE_MASK_BITS_MAX     CORE_MASK_BITS_ON_ICX
 #define CHAN_RANK_MAX          CHAN_RANK_MAX_ON_HSX
 #define DIMM_IDX_MAX           DIMM_IDX_MAX_ON_HSX
-- 
2.25.1

